firebase-firestore/src/main/java/com/google/firebase/firestore/Pipeline.kt

```diff
@@ -0,0 +1,1205 @@
+   *   .aggregate(
+   *     Aggregate
+   *       .withAccumulators(average("rating").as("avg_rating"))
+   *       .withGroups("genre"));
+   * ```
+   *
+   * @param aggregateStage An [AggregateStage] object that specifies the grouping fields (if any)
+   * and the aggregation operations to perform.
+   * @return A new [Pipeline] object with this stage appended to the stage list.
+   */
+  fun aggregate(aggregateStage: AggregateStage): Pipeline =
+    aggregate(aggregateStage, AggregateOptions())
+
+  /**
+   * Performs optionally grouped aggregation operations on the documents from previous stages.
+   *
+   * This stage allows you to calculate aggregate values over a set of documents, optionally grouped
+   * by one or more fields or functions. You can specify:
+   *
+   * - **Grouping Fields or Expressions:** One or more fields or functions to group the documents
+   * by. For each distinct combination of values in these fields, a separate group is created. If no
+   * grouping fields are provided, a single group containing all documents is used. Not specifying
+   * groups is the same as putting the entire inputs into one group.
+   *
+   * - **AggregateFunctions:** One or more accumulation operations to perform within each group.
+   * These are defined using [AliasedAggregate] expressions, which are typically created by calling
+   * [AggregateFunction.alias] on [AggregateFunction] instances. Each aggregation calculates a value
+   * (e.g., sum, average, count) based on the documents within its group.
+   *
+   * @param aggregateStage An [AggregateStage] object that specifies the grouping fields (if any)
+   * and the aggregation operations to perform.
+   * @param options The [AggregateOptions] to use when performing the aggregation.
+   * @return A new [Pipeline] object with this stage appended to the stage list.
+   */
+  fun aggregate(aggregateStage: AggregateStage, options: AggregateOptions): Pipeline =
+    append(aggregateStage.withOptions(options))
+
+  /**
+   * Performs a vector similarity search, ordering the result set by most similar to least similar,
+   * and returning the first N documents in the result set.
+   *
+   * @param vectorField A [Field] that contains vector to search on.
+   * @param vectorValue The [VectorValue] in array form that is used to measure the distance from
+   * [vectorField] values in the documents.
+   * @param distanceMeasure specifies what type of distance is calculated when performing the
+   * search.
+   * @return A new [Pipeline] object with this stage appended to the stage list.
+   */
+  fun findNearest(
+    vectorField: Field,
+    vectorValue: DoubleArray,
+    distanceMeasure: FindNearestStage.DistanceMeasure,
+  ): Pipeline = append(FindNearestStage.of(vectorField, vectorValue, distanceMeasure))
+
+  /**
+   * Performs a vector similarity search, ordering the result set by most similar to least similar,
+   * and returning the first N documents in the result set.
```

Comment:
This overload will return the full result set.

Reply:
Fixed.